tests/test-repeat-posts.php

```diff
@@ -200,23 +204,50 @@ function test_create_repeat_unsupported_repeating_post_type() {
 
 	}
 
+	/**
+	 * TODO: This test needs re-writing!
+	 *
+	 * The code is using $_POST global as a deciding condition where to save a meta data for a post or not.
+	 * This test doesn't use $_POST, but sets meta data via separate `save_post_repeating_status` function.
+	 *
+	 * So, the live and test environment have setup differences - which previously
+	 * resulted in the fact that the test did pass, while in reality the `hm-post-repeat`
+	 * meta data was created for a Repeat post (while it shouldn't have).
+	 *
+	 * That issue is fixed in this current PR, but I'm struggling to correctly test it.
+	 * The difficulty is due to `save_post` hook and therefore 2 functions hooked into it,
+	 * being triggered again when a Repeat post is being `wp_insert_post`. Thus resulting
+	 * in code inception - that in turn uses $_POST global as a deciding condition whether
+	 * to save a meta data or not.
+	 *
+	 * We need to replicate a live environment conditions with $_POST global.
```

Comment:
I haven’t fully grasped this, but could we just manually set the `$_POST` superglobal match the live environment?